src/main/java/catan/services/util/play/ActionParamsUtil.java

```diff
@@ -0,0 +1,125 @@
+import static catan.domain.model.dashboard.types.NodeBuiltType.SETTLEMENT;
+
+@Component
+public class ActionParamsUtil {
+    private Logger log = LoggerFactory.getLogger(ActionParamsUtil.class);
+
+    public List<Integer> calculateBuildSettlementParams(GameBean game, GameUserBean gameUser) {
+
+        if (game.getStage().equals(PREPARATION)) {
+            return fetchNodeIdsToBuildOnInPreparationStage(game);
+        }
+
+        List<Integer> nodeIdsToBuildOn = new ArrayList<Integer>();
+        for (NodeBean node : game.getNodes()) {
+            if (node.getBuilding() == null
+                    && node.noBuildingsOnNeighbourNodes()
+                    && node.nearGameUsersNeighbourRoad(gameUser)) {
+                nodeIdsToBuildOn.add(node.getId());
+            }
+        }
+        return nodeIdsToBuildOn;
+    }
+
+    public List<Integer> calculateBuildCityParams(GameBean game, GameUserBean gameUser) {
+
+        if (game.getStage().equals(PREPARATION)) {
+            return fetchNodeIdsToBuildOnInPreparationStage(game);
+        }
+
+        List<Integer> nodeIdsToBuildOn = new ArrayList<Integer>();
+        for (NodeBean node : game.getNodes()) {
+            if (node.getBuilding() != null
+                    && node.getBuilding().getBuildingOwner().equals(gameUser)
+                    && node.getBuilding().getBuilt().equals(SETTLEMENT)) {
+                nodeIdsToBuildOn.add(node.getId());
+            }
+        }
+        return nodeIdsToBuildOn;
+    }
+
+    public List<Integer> calculateBuildRoadParams(GameBean game, GameUserBean gameUser) {
+        List<Integer> edgeIdsToBuildOn = new ArrayList<Integer>();
+        if (game.getStage().equals(PREPARATION)) {
+            for (NodeBean node : game.getNodes()) {
+                if (node.getBuilding() != null
+                        && node.getBuilding().getBuildingOwner().equals(gameUser)
+                        && !node.nearGameUsersNeighbourRoad(gameUser)) {
+                    for (EdgeBean edge : node.getEdges().listAllNotNullItems()) {
+                        edgeIdsToBuildOn.add(edge.getId());
+                    }
+                    break;
+                }
+            }
+            return edgeIdsToBuildOn;
+        }
+
+        for (EdgeBean edge : game.fetchEdgesAccessibleForBuildingRoadInMainStage(gameUser)) {
+            edgeIdsToBuildOn.add(edge.getId());
+        }
+        return edgeIdsToBuildOn;
+    }
+
+    private List<Integer> fetchNodeIdsToBuildOnInPreparationStage(GameBean game) {
+        List<Integer> nodeIdsToBuildOn = new ArrayList<Integer>();
+        for (NodeBean node : game.getNodes()) {
+            if (node.getBuilding() == null && node.noBuildingsOnNeighbourNodes()) {
+                nodeIdsToBuildOn.add(node.getId());
+            }
+        }
+        return nodeIdsToBuildOn;
+    }
+
+    public ResourcesParams calculateResourcesParams(GameUserBean gameUser, GameBean game) {
```

Comment:
move this method up to other puclic methods, rename it to calculateTradePortParams